evm-contracts/src/v0.6/dev/FluxAggregator.sol

```diff
@@ -609,71 +615,76 @@ contract FluxAggregator is AggregatorInterface, Owned {
    * Private
    */
 
-  function initializeNewRound(uint32 _id)
+  function initializeNewRound(uint32 _roundId)
     private
   {
-    updateTimedOutRoundInfo(_id.sub(1));
+    updateTimedOutRoundInfo(_roundId.sub(1));
 
-    reportingRoundId = _id;
-    rounds[_id].details.maxAnswers = maxAnswerCount;
-    rounds[_id].details.minAnswers = minAnswerCount;
-    rounds[_id].details.paymentAmount = paymentAmount;
-    rounds[_id].details.timeout = timeout;
-    rounds[_id].startedAt = uint64(block.timestamp);
+    reportingRoundId = _roundId;
+    rounds[_roundId].details.maxAnswers = maxAnswerCount;
+    rounds[_roundId].details.minAnswers = minAnswerCount;
+    rounds[_roundId].details.paymentAmount = paymentAmount;
+    rounds[_roundId].details.timeout = timeout;
+    rounds[_roundId].startedAt = uint64(block.timestamp);
 
-    emit NewRound(_id, msg.sender, rounds[_id].startedAt);
+    emit NewRound(_roundId, msg.sender, rounds[_roundId].startedAt);
   }
 
-  function oracleInitializeNewRound(uint32 _id)
+  function oracleInitializeNewRound(uint32 _roundId)
     private
-    ifNewRound(_id)
-    ifOracleDelayed(_id)
   {
-    initializeNewRound(_id);
+    if (!newRound(_roundId)) return;
+    uint256 lastStarted = oracles[msg.sender].lastStartedRound; // cache storage reads
+    if (_roundId <= lastStarted + restartDelay && lastStarted != 0) return;
```

Comment:
It is necessary that we don't revert here, because a lot of `updateAnswer` follows after it calls `oracleInitializeNewRound`. We could move the `if` check into `updateAnswer` to make that more clear. This is sort of an artifact of the modifier driven flow before.